### PR TITLE
fix(#6177): improve about page for mobile readability

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

[description]

medic/cht-core#6177

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:6177-about-page-for-mobile/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:6177-about-page-for-mobile/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:6177-about-page-for-mobile/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
@latin-panda This page is not done, but hopefully this is an incremental improvement which at least means all the information is readable. If you're happy I'll split the issue into two - one for this incremental improvement and one for a rethink of content.

**Review by latin-panda — CHANGES_REQUESTED** [review]
Nice! I left some recommendations inline

File: tests/page-objects/default/about/about.wdio.page.js
Comment (garethbowen) [reply tests/page-objects/default/about/about.wdio.page.js:3]: I tend to use IDs only as a last resort, for two reasons. Firstly, it's adding production code purely for testing which isn't ideal. Secondly, this is actually testing the user facing text - the user doesn't care what the ID is but if we change the label to something else we want the test to fail because we've broken the user experience.

Why do you prefer adding IDs?

File: webapp/src/ts/modules/about/about.component.html
Comment (garethbowen) [reply webapp/src/ts/modules/about/about.component.html:41]: Something got lost in translation... I think you're suggesting removing the `<div>`s?

I think with the current CSS they're needed to fire this line: https://github.com/medic/cht-core/blob/master/webapp/src/css/material.less#L124

This is reusing the same structure as this code: https://github.com/medic/cht-core/blob/master/webapp/src/ts/modules/contacts/contacts-content.component.html#L61

File: tests/page-objects/default/about/about.wdio.page.js
Comment (latin-panda) [reply tests/page-objects/default/about/about.wdio.page.js:3]: We started using this since the protractor times as a way to identify content, especially dynamic content like contact summaries; otherwise, it would be hard for the e2e to find the correct piece of information to verify correctness. We need to provide enough identifiable elements to facilitate the e2e. 

In this case, the About page isn't dynamic because the labels will always be in the same position, but breaking the e2e test because the element changed position or type is rarely helpful, and what matters the most is that the data is displayed. 

This also comes in handy when the selectors are too complex or inefficient (selectors can also impact performance). Something like this makes it cleaner and faster for those with less experience.  

The `test-id` is harmless and ignored by the browser and processors, so adding it or not, doesn't make a difference for the end user in production. 

In this case, checking that the data is there is the important piece, not the position, so I suggested this to avoid breaking the e2e because of a change in the DOM.

File: webapp/src/ts/modules/about/about.component.html
Comment (latin-panda) [reply webapp/src/ts/modules/about/about.component.html:41]: Ahh, that's because `contacts-content.component.html#L61` has a label before the `div` and helps to properly align the other elements in relation to that label. Be here we don't have that, so looks like an unnecessary extra structure. 

I think it's safer to use the regular [MatGrid](https://material.angular.io/components/grid-list/overview). The structure is standard, fast to get it to work, less likely to get overridden by accident, and easier to understand. Because this other thing is a custom arrangement from contact-content, it needs more digging before we can do maintenance work.

File: tests/page-objects/default/about/about.wdio.page.js
Comment (garethbowen) [reply tests/page-objects/default/about/about.wdio.page.js:3]: > We started using this since the protractor times as a way to identify content,

I agree that with dynamic content it can be difficult to find the exact element you want. This is the last resort where I would use IDs. But as you say, doesn't apply in this case.

> (selectors can also impact performance). 

Not a concern in e2e testing. We have many places where we literally pause the run for seconds at a time so a microsecond of selector performance won't make a difference.

> The test-id is harmless and ignored by the browser and processors

It adds clutter to the template which makes the code harder to maintain.

> In this case, checking that the data is there is the important piece, not the position

I disagree. If the data shows up with a different label then the data is wrong, for example, if we started showing "latin-panda" underneath "supported browser" then that should result in a failing test. With the ID approach it'll pass.

File: webapp/src/ts/modules/about/about.component.html
Comment (garethbowen) [reply webapp/src/ts/modules/about/about.component.html:41]: > Ahh, that's because contacts-content.component.html#L61 has a label before the div and helps to properly align the other elements in relation to that label. Be here we don't have that, so looks like an unnecessary extra structure.

Yeah exactly.

MatGrid seems overkill as this is mostly a vertical list not a grid (the data usage part is a grid but I haven't modified that at all). Wouldn't a [list](https://m3.material.io/components/lists/overview) be more fitting?

Comment (latin-panda) [reply webapp/src/ts/modules/about/about.component.html:41]: I was thinking the real state suggestion from Nicole, MatGrid makes columns easier to do. The [MatList](https://material.angular.io/components/list/overview) also works

File: tests/page-objects/default/about/about.wdio.page.js
Comment (latin-panda) [reply tests/page-objects/default/about/about.wdio.page.js:3]: >  If the data shows up with a different label then the data is wrong

Then we need to add tests to the labels to know if they change places by mistake 😬 it's a data pair, and it seems reasonable to verify them.

IMHO, the gains are higher than the consequences of adding `test-id`, mainly because e2e tests have become an essential piece in our development experience. But we add them only in specific scenarios where selectors are complex or content is too dynamic. 

I'm not strongly opinionated for the About page, and it's okay to drop the suggestion.

File: webapp/src/ts/modules/about/about.component.html
Comment (garethbowen) [reply webapp/src/ts/modules/about/about.component.html:41]: MatGrid is no good because the second column isn't responsive so we'd end up with not enough horizontal space on mobile again.

I think each section should be a Card with a List inside.

Comment (garethbowen) [reply webapp/src/ts/modules/about/about.component.html:41]: What a nightmare. It turns out List doesn't expand to fit the content, it just truncates with an ellipsis, and it's impossible to overwrite because it fixes the element height to the number of lines it expects. In other words it's not responsive at all.

Cards are pretty but also not responsive, so if we want a grid of cards we're going to have to implement it ourselves anyway.

File: tests/page-objects/default/about/about.wdio.page.js
Comment (garethbowen) [reply tests/page-objects/default/about/about.wdio.page.js:15]: Done.

File: webapp/src/ts/modules/about/about.component.html
Comment (garethbowen) [reply webapp/src/ts/modules/about/about.component.html]: I've gone for cards but ignored responsive design and gone for mobile first instead.

**Comment by garethbowen** [comment]
@latin-panda Can I get another review please?

File: webapp/tests/karma/ts/modules/about/about.component.spec.ts
Comment (latin-panda) [inline webapp/tests/karma/ts/modules/about/about.component.spec.ts:11]:
```suggestion
import { MatCardModule } from '@angular/material/card';
import { expect } from 'chai';
import sinon from 'sinon';

import { AboutComponent } from '@mm-modules/about/about.component';
import { ResourceIconsService } from '@mm-services/resource-icons.service';
```

File: webapp/src/ts/modules/about/about.component.html
Comment (latin-panda) [inline webapp/src/ts/modules/about/about.component.html:1]: This line adds unnecessary gray line


It looks better without `<div class="row filters"></div>`

**Review by latin-panda — APPROVED** [review]
Nice! Just a couple of minor comments but approving to unblock

**Pull request merged**
